### PR TITLE
1.9: fix #32: linear brightness

### LINKED ISSUE #32: Could you implement a software brightness feature?
For example I use xrandr to change brightness:

xrandr --output DVI-D-0 --brightness 0.6

xrandr --output DVI-D-0 --brightness 1.0

however sct simply overrides the software brightness. Possibly could you either make these not impede eachother, or integrate that functionality?
For example, gummy has similar functionality:
https://github.com/?q=gummy
however he doesn't have support for fedora currently.

### PR DESCRIPTION
**Pull request by zvezdochiot**
Add second argument: Linear brightness

Use:
```sh
./xsct 4000 0.5 && ./xsct
```

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,5 +1,8 @@
 PROJECT: https://github.com/faf0/sct
 
+1.9: zvezdochiot on 11 Aug 2022
```
Comment (faf0) [inline CHANGELOG]: should this be updated to
```suggestion
1.9: zvezdochiot on 15 Dec 2022
```
?

File: README.md [hunk 2]
```diff
@@ -80,14 +80,16 @@ gcc -Wall -Wextra -Werror -pedantic -std=c99 -O2 -I /usr/X11R6/include xsct.c -o
 
 Provided that xsct binary is located in your `$PATH`, execute it using the following command:
 ~~~sh
-xsct 3700
+xsct 3700 0.9
 ~~~
 
 The first parameter (`3700` above) represents the color temperature.  
 
+The second parameter (`0.9` above) represents the brightness.
```
Comment (faf0) [inline README.md]:
```suggestion
The second parameter (`0.9` above) represents the brightness. The values are in the range `[0.0, 1.0]`.
```

File: src/xsct.c [hunk 3]
```diff
@@ -4,70 +4,46 @@
...
     XRRScreenResources *res = XRRGetScreenResourcesCurrent(dpy, root);
 
-    int temp = 0, n, c;
+    int n, c;
     double t = 0.0;
-    double gammar = 0.0, gammag = 0.0, gammab = 0.0, gammam = 1.0, gammad = 0.0;
+    double gammar = 0.0, gammag = 0.0, gammab = 0.0, gammad = 0.0;
+    struct temp_status temp;
+    temp.temp = 0;
+    temp.brightness = 1.0;
```
Comment (faf0) [inline src/xsct.c:39]: note: this variable replaces `gammam`.

File: src/xsct.c [hunk 4]
```diff
@@ -107,34 +86,37 @@ static int get_sct_for_screen(Display *dpy, int screen, int icrtc, int fdebug)
...
+    temp.temp = (int)(t + 0.5);
 
     return temp;
 }
 
-static void sct_for_screen(Display *dpy, int screen, int icrtc, int temp, int fdebug)
+static void sct_for_screen(Display *dpy, int screen, int icrtc, struct temp_status temp, int fdebug)
 {
-    double t = 0.0, g = 0.0, gammar, gammag, gammab;
+    double t = 0.0, b = 1.0, g = 0.0, gammar, gammag, gammab;
```
Comment (faf0) [inline src/xsct.c:99]: note: `b` represents the brightness

File: xsct.1 [hunk 5]
```diff
@@ -32,7 +33,10 @@ Black body temperature
 .br
 If the value is 0, xsct sets the color temperature to the default of 6500
 .br
-If no arguments are passed, xsct estimates the current display temperature
+If no arguments are passed, xsct estimates the current display temperature and brightness
+.TP
+.I [brightness]
+Linear brightness {0.0 .. 1.0}
```
Comment (faf0) [inline xsct.1]:
```suggestion
Linear brightness between 0.0 (inclusive) and 1.0 (inclusive). The default value is 1.0 if no parameter is passed.
```

**Comment by faf0** [comment]
Thanks, @zvezdochiot ! Look good. Only thing left to do is to update the documentation.

**Comment by zvezdochiot** [comment]
@faf0 say:
>update the documentation.

Resolved in https://github.com/faf0/sct/pull/33/commits/93dec592fbe006b4766eecbfe2c727ab796f7509 .

File: README.md [hunk 6]
```diff
@@ -80,14 +80,16 @@ gcc -Wall -Wextra -Werror -pedantic -std=c99 -O2 -I /usr/X11R6/include xsct.c -o
...
-The first parameter (`3700` above) represents the color temperature.  
+The first parameter (`3700` above) represents the color temperature.
 
-If `xsct` is called with parameter 0, the color temperature is set to `6500`.  
+The second parameter (`0.9` above) represents the brightness. The values are in the range `[0.0, 1.0]`.
 
-If `xsct` is called without parameters, the current display temperature is estimated.
+If `xsct` is called with parameter 0, the color temperature is set to `6500`.
+
+If `xsct` is called without parameters, the current display temperature and brightness is estimated.
```
Comment (faf0) [inline README.md]:
```suggestion
If `xsct` is called without parameters, the current display temperature and brightness are estimated.
```

File: README.md [hunk 7]
```diff
@@ -80,14 +80,16 @@ gcc -Wall -Wextra -Werror -pedantic -std=c99 -O2 -I /usr/X11R6/include xsct.c -o
...
 
-The first parameter (`3700` above) represents the color temperature.  
+The first parameter (`3700` above) represents the color temperature.
 
-If `xsct` is called with parameter 0, the color temperature is set to `6500`.  
+The second parameter (`0.9` above) represents the brightness. The values are in the range `[0.0, 1.0]`.
 
-If `xsct` is called without parameters, the current display temperature is estimated.
+If `xsct` is called with parameter 0, the color temperature is set to `6500`.
+
```
Comment (faf0) [inline README.md:91]:
```suggestion

If `xsct` is called with the color temperature parameter only, the brightness is set to `1.0`.

```

**Comment by faf0** [comment]
> @faf0 say:
> 
> > update the documentation.
> 
> Resolved in [93dec59](https://github.com/faf0/sct/commit/93dec592fbe006b4766eecbfe2c727ab796f7509) .

Thanks again! After two small tweaks to the README.md, this is good to merge.

**Comment by faf0** [comment]
Addresses https://github.com/faf0/sct/issues/32.

**Comment by zvezdochiot** [comment]
@faf0 say:
>two small tweaks to the README.md

Resolved in https://github.com/faf0/sct/pull/33/commits/12cb2ab95490dd99a5ca81f8449721b6a4509571.

**Review by faf0 — APPROVED** [review]
Thanks, @zvezdochiot !

**Pull request merged**